cfme/services/catalogs/catalog_item.py

```diff
@@ -1,107 +1,202 @@
+from utils import version, deferred_verpick
+
+from . import ServicesCatalogView
+
+
+class BasicInfoForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name='description')
+    display = Checkbox(name='display')
+    select_catalog = BootstrapSelect('catalog_id')
+    select_dialog = BootstrapSelect('dialog_id')
+    select_orch_template = BootstrapSelect('template_id')
+    select_provider = BootstrapSelect('manager_id')
+    select_config_template = BootstrapSelect('template_id')
+    field_entry_point = Input(name='fqname')
+    retirement_entry_point = Input(name='retire_fqname')
+    select_resource = BootstrapSelect('resource_id')
+    tree = deferred_verpick({version.LOWEST: ManageIQTree('automate_treebox'),
+         '5.7': BootstrapTreeview('automate_treebox')})
+    cancel_button = Button('Cancel')
+
+
+class ButtonGroupForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    btn_group_text = Input(name='name')
+    btn_group_hvr_text = Input(name='description')
+    btn_image = BootstrapSelect('button_image')
+
+
+class ButtonForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    btn_text = Input(name='name')
+    btn_hvr_text = Input(name='description')
+    select_dialog = BootstrapSelect('dialog_id')
+    system_process = BootstrapSelect('instance_name')
+    request = Input(name='object_request')
+    btn_image = BootstrapSelect('button_image')
+
+
+class EditTagsForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    select_tag = BootstrapSelect('tag_cat')
+    select_value = BootstrapSelect('tag_add')
+
+
+class AllCatalogItemView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            self.title.text == 'All Service Catalog Items' and
+            self.catalog_items.is_opened and
+            self.catalog_items.tree.currently_selected == ["All Catalog Items"])
+
+
+class DetailsCatalogItemView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.catalog_items.is_opened and
+            self.title.text == 'Service Catalog Item "{}"'.format(self.context['object'].name)
+        )
+
+class CatalogForm(BasicInfoForm):
+    select_item_type = BootstrapSelect('st_prov_type')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and self.catalog_items.is_opened and
+            self.title.text == "Adding a new Service Catalog Item"
+        )
+
+    def before_fill(self, values):
+        item_type = self.context['object'].provider_type or \
+            self.context['object'].item_type or 'Generic'
+        sel.select("//select[@id='st_prov_type']", item_type)
+        # self.select_item_type.select_by_visible_text(item_type)
```

Comment:
No , I used the commented line and got the error . This is to show that error doesn't come if I use sel.select